src/test/java/net/minecraftforge/debug/CustomSlimeBlockTest.java

```diff
@@ -0,0 +1,127 @@
+import net.minecraftforge.client.event.ModelRegistryEvent;
+import net.minecraftforge.client.model.ModelLoader;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.Mod.EventBusSubscriber;
+import net.minecraftforge.fml.common.Mod.EventHandler;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+@Mod(modid = CustomSlimeBlockTest.MODID, name = "ForgeDebugCustomSlimeBlock", version = CustomSlimeBlockTest.VERSION, acceptableRemoteVersions = "*")
+public class CustomSlimeBlockTest
+{
+    public static final String MODID = "forgedebugcustomslimeblock";
+    public static final String ASSETS = "forgedebugcustomslimeblock:";
+    public static final String VERSION = "1.0";
+    public static Block CUSTOM_SLIME_BLOCK;
+
+    @EventHandler
+    public void preInit(FMLPreInitializationEvent event)
+    {
+        CUSTOM_SLIME_BLOCK = new CustomSlime();
+        ForgeRegistries.BLOCKS.register(CUSTOM_SLIME_BLOCK);
+        ForgeRegistries.ITEMS.register(new ItemMultiTexture(CUSTOM_SLIME_BLOCK, CUSTOM_SLIME_BLOCK, stack -> CustomSlime.BlockType.values[stack.getMetadata()].toString()).setRegistryName(CUSTOM_SLIME_BLOCK.getRegistryName()));
+    }
+
+    @EventBusSubscriber(value = Side.CLIENT, modid = MODID)
+    public static class BakeEventHandler
+    {
+        @SubscribeEvent
+        public static void registerModels(ModelRegistryEvent event)
+        {
+            Item item = Item.getItemFromBlock(CUSTOM_SLIME_BLOCK);
+            ModelLoader.setCustomModelResourceLocation(item, 0, new ModelResourceLocation(ASSETS + "blue_slime_block", "inventory"));
+            ModelLoader.setCustomModelResourceLocation(item, 1, new ModelResourceLocation(ASSETS + "purple_slime_block", "inventory"));
+            ModelLoader.setCustomModelResourceLocation(item, 2, new ModelResourceLocation(ASSETS + "obsidian_slime_block", "inventory"));
+        }
+    }
+
+    public static class CustomSlime extends BlockSlime
+    {
+        private static final PropertyEnum<BlockType> VARIANT = PropertyEnum.create("variant", BlockType.class);
+
+        private CustomSlime()
+        {
+            super();
+            this.setDefaultState(this.blockState.getBaseState().withProperty(VARIANT, BlockType.BLUE_SLIME_BLOCK));
+            this.setCreativeTab(CreativeTabs.DECORATIONS);
+            this.setUnlocalizedName("custom_slime_block");
+            this.setRegistryName("forgedebugcustomslimeblock:custom_slime_block");
+            this.setSoundType(SoundType.SLIME);
+        }
+
+        @Override
+        @SideOnly(Side.CLIENT)
+        public void getSubBlocks(CreativeTabs creativeTabs, NonNullList<ItemStack> list)
+        {
+            for (int i = 0; i < BlockType.values.length; ++i)
```

Comment:
Synthetic access. `values` field should be package-protected instead.
https://stackoverflow.com/a/1801801

Reply:
Done.